src/transform.ts

```diff
@@ -5,119 +5,132 @@ import * as JSON5 from "json5";
 const packageName = "env0-ts-transform-json-schema";
 
 export interface TransformerOptions {
-  env: { [key: string]: string };
+    env: { [key: string]: string };
+}
+
+const getTypeName = ({right, left, escapedText}) => {
+    /* right is the current type name
+    * left is the father
+    * this way we build the type name recursively
+    * if we don't have any father then we take the escapedText of the current node which is the Root
+     */
+    if (!right || !left) {
+        return escapedText;
+    }
+    return `${getTypeName(left)}.${right.escapedText}`
 }
```

Comment:
1. I'm not sure I got the explanation here but if I did, consider aliasing your parameters instead of having mapping comment.  

2. The purpose of this function is still to get a fully qualified name, right?  

3. Consider an iterative approach over a recursive one and don't handle both the qualifier and node in same pass

```suggestion
const getFullyQualifiedTypeName = ({ right: node, left: qualifier }) => {
  const qualifiers = []
  
  while(node) {
    qualifiers.push(node.escapedText);
    node = qualifier;
  }

  return qualifiers.reverse().join('.');
}
```

Reply:
1. i can change them to `current` and `parent`
2. yes, it builds the type hierarchy
3. I'll try to work with what you suggested

Reply:
It's not "parent" though - it's qualifier, isn't it? Like `Foo.Bar.Moo` is the fully qualified name for `Moo`.

Reply:
d87d9db1

But we get each one separately meaning we first running on `Moo` then `node.left` bring us `Bar`

Reply:
Yes, which is why `left` is the qualifier that's how qualifiers work - it's the thing before the previous dot.